applications/MeshMovingApplication/python_scripts/mesh_solver_base.py

```diff
@@ -94,54 +123,47 @@ def InitializeSolutionStep(self):
     def FinalizeSolutionStep(self):
         self.get_mesh_motion_solving_strategy().FinalizeSolutionStep()
 
+    def Predict(self):
+        self.get_mesh_motion_solving_strategy().Predict()
+
+    def SolveSolutionStep(self):
+        self.get_mesh_motion_solving_strategy().Solve() # Calling Solve bcs this is what is currently implemented in the MeshSolverStrategies
+
     def SetEchoLevel(self, level):
         self.get_mesh_motion_solving_strategy().SetEchoLevel(level)
 
     def GetEchoLevel(self):
         self.get_mesh_motion_solving_strategy().GetEchoLevel()
 
-    def Solve(self):
-        self.get_mesh_motion_solving_strategy().Solve()
-
     def Clear(self):
         self.get_mesh_motion_solving_strategy().Clear()
 
     def Check(self):
         self.get_mesh_motion_solving_strategy().Check()
 
+    def GetMinimumBufferSize(self):
+        time_order = self.settings["time_order"].GetInt()
```

Comment:
same comment as above, i would have the user to prescribe how mesh velocity is to be treated

Reply:
you are right, see my answer above